tensorflow/python/ops/image_ops_test.py

```diff
@@ -1279,6 +1279,29 @@ def testShape(self):
                          [None, None, channels or None])
 
 
+class GifTest(test_util.TensorFlowTestCase):
+
+  def testExisting(self):
+    # Read some real GIFs, converting to different channel numbers
+    prefix = 'tensorflow/core/lib/gif/testdata/'
+    inputs = (3, 'cock.gif'),
+    for channels_in, filename in inputs:
+      for channels in 0, 1, 3, 4:
+        with self.test_session() as sess:
+          gif0 = io_ops.read_file(prefix + filename)
+          image0 = image_ops.decode_gif(gif0, channels=channels)
+          gif0, image0 = sess.run([gif0, image0])
+          self.assertEqual(image0.shape, (55, 63, 3))
```

Comment:
Can you add some tests to make sure the contents of the read gif is what we would expect?